WolvenKit.RED4.TweakDB/Types/TweakDBID.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.IO;
+using RED.CRC32;
+using System.Text.RegularExpressions;
+using System.Buffers.Binary;
+using WolvenKit.RED4.CR2W.Types;
+
+namespace WolvenKit.RED4.TweakDB.Types
+{
+    public class TweakDBID : IPrimitive
+    {
+        public string Name => "TweakDBID";
+        public string Text { get; set; }
+        public int Length { get; set; }
+        public uint Key { get; set; }
+        public static implicit operator TweakDBID(string value) {
+            var t = new TweakDBID();
+            var rx = new Regex(@"Tw?e?a?k?DBID\w?:\w?0?x?([0-9A-F]{8})\w?:\w?0?x?([0-9A-F]{2})", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+            var m = rx.Matches(value);
+            if (m.Count == 1)
+            {
+                t.Text = value;
+                t.Key = Convert.ToUInt32(m[0].Groups[1].Value, 16);
+                t.Length = Convert.ToUInt16(m[0].Groups[2].Value, 16);
+                return t;
+            }
+
+            t.Text = value;
+            t.Key = Crc32Algorithm.Compute(value);
+            t.Length = value.Length;
+            return t;
+        }
+        public override string ToString() => $"{Text} <TweakDBID 0x{String.Format("{0:X8}", Key)}:0x{String.Format("{0:X2}", Length)} / {Key}:{Length}>";
```

Comment:
New line before this, and let's use string interpolation, no need for `String.Format`.

Reply:
Ah, sorry - I thought I got this changed, but I missed the `String.Format` somehow :/